src/basic/unit-file.c

```diff
@@ -586,17 +586,21 @@ int unit_file_build_name_map(
         /* Let's also put the names in the reverse db. */
         const char *dummy, *src;
         HASHMAP_FOREACH_KEY(dummy, src, ids) {
-                _cleanup_free_ char *inst = NULL, *dst_inst = NULL;
-                const char *dst;
+                _cleanup_free_ char *inst = NULL, *dst_inst = NULL, *dst = NULL;
+                const char *dst_path;
 
-                r = unit_ids_map_get(ids, src, &dst);
+                r = unit_ids_map_get(ids, src, &dst_path);
                 if (r < 0)
                         continue;
 
-                if (null_or_empty_path(dst) != 0)
+                if (null_or_empty_path(dst_path) != 0)
                         continue;
 
-                dst = basename(dst);
+                r = path_extract_filename(dst_path, &dst);
+                if (r < 0) {
+                        log_debug_errno(r, "Failed to extract basename from %s: %m", dst);
```

Comment:
we generally want that if we ignore some error condition that our log message says so. i.e. insert ", ignoring" into your log message:

```
log_debug_errno(r, "Failed to extract basename from %s, ignoring: %m", dst);
```

i.e. we typically want that our log message that might or might indicate a problem still indicate what action is being taken now, and that includes if the error is ignored.